src/pages/Topup.vue

```diff
@@ -45,35 +45,51 @@ export default {
   data() {
     return {
       giftCardCode: '',
-      openedScanner: false,
-      currentQRCodeScanner: null,
+
+      scannerStarted: false,
+      theScanner: null,
     }
   },
   computed: {
     ...mapGetters('customer', ['getCustomer']),
+    haveGiftCardCode() {
+      return this.giftCardCode !== ''
+    },
+    colorGiftCardCode() {
+      return this.giftCardCode === '' ? 'tertiary' : 'tertiarylight'
+    },
+    classGiftCardCode() {
+      return this.giftCardCode === '' ? 'q-ma-sm col-11' : 'q-ma-sm col-11 text-secondary'
+    },
   },
   methods: {
     ...mapActions('giftcard', ['updateGiftCard']),
+    clearGiftCardCode() {
+      this.giftCardCode = ''
+    },
+    setGiftCardCode(code) {
+      setTimeout(() => {
```

Comment:
setTimeout v-> for consistent performance -> might consider to use async/await ?
coz from that fact that we need to wait for something, we could consider it to be a asynced-process